bundles/org.eclipse.jface/src/org/eclipse/jface/resource/URLImageDescriptor.java

```diff
@@ -215,12 +226,49 @@ private static URL getxURL(URL url, int zoom) {
 				if (url.getQuery() != null) {
 					file += '?' + url.getQuery();
 				}
-				return new URL(url.getProtocol(), url.getHost(), url.getPort(), file);
+				resultingURL = new URL(url.getProtocol(), url.getHost(), url.getPort(), file);
+			} catch (MalformedURLException e) {
+				Policy.getLog().log(new Status(IStatus.ERROR, Policy.JFACE, e.getLocalizedMessage(), e));
+			}
+		}
+
+		// Next we check if there's an alternate image that we should use
+		// TODO: I'm not exactly sure how we should be accessing the JFace preference
+		// Should a IPreferenceStore instance be retrieved?
+		// Or is that considered part of Eclipse RCP and not JFace?
+		if (
+		// !JFaceResources.getString(JFacePreferences.CUSTOM_RESOURCE_THEME).equals(JFacePreferences.CUSTOM_RESOURCE_THEME)
+		// &&
+		resultingURL.getFile().contains(PLUGIN_URL)) {
+			// The third path segment is the bundleID
+			String originalFilePath = resultingURL.getFile();
+			for (int i = 0; i < 2; i++) {
+				originalFilePath = originalFilePath.substring(originalFilePath.indexOf('/') + 1);
+			}
+			// TODO: The original bundleID should only be used in the case of plugin
+			// fragment image replacements.
+			// We need an additional (optional) preference that will hold the bundleID of
+			// the currently used theme plugin
+			String bundleID = originalFilePath.substring(0, originalFilePath.indexOf('/'));
+			originalFilePath = originalFilePath.substring(originalFilePath.indexOf('/'), originalFilePath.length());
+
+			// TODO: This is a workaround and should probably be removed
+			if (originalFilePath.contains("/$nl$/")) { //$NON-NLS-1$
+
+				originalFilePath = originalFilePath.replace("/$nl$/", "/"); //$NON-NLS-1$ //$NON-NLS-2$
```

Comment:
Why is `$nl$` removed here? I think it should be kept and resolved as usual without particular case.